### PR TITLE
control programmatically if validation errors should be shown

### LINKED ISSUE #364: Make bs-form showAllValidations property public?
I've sometimes uses cases where I have to show all validation errors independently of form submission. `showAllValidations` property of `{{bs-form}}` is very helpful therefore. But it's currently private. Is there any reason why it shouldn't be public? Or is there a better way to achieve that?

### PR DESCRIPTION
**Pull request by dagroe**
Adds a public property `@showValidations` to `<BsForm>` to show/hide validations as proposed by @jelhan in https://github.com/ember-bootstrap/ember-bootstrap/issues/364#issuecomment-309417556

close #364

### CODE REVIEW

**Review by simonihmig — APPROVED** [review]
LGTM! Thanks @dagroe!

@jelhan can I have your 👀 also here, as this API was proposed by you in the related issue?

**Comment by simonihmig** [comment]
And sorry for the late review! 😔

File: tests/integration/components/bs-form-test.js
Comment (jelhan) [inline tests/integration/components/bs-form-test.js:324]: Would be great if you could extend the test to verify that success validation state is also shown.

File: tests/integration/components/bs-form-test.js
Comment (jelhan) [inline tests/integration/components/bs-form-test.js:342]: Similar as above, it would be great if you could extend the test to verify that also success validation is not shown.

**Review by jelhan — CHANGES_REQUESTED** [review]
I'm sorry for the late review. I started the review some time ago but missed to submit it.

**Comment by joeo73-impinj** [comment]
I would approve also if I had permission, I could really use this feature! :-)

File: addon/components/bs-form.js
Comment (dagroe) [reply addon/components/bs-form.js]: done

File: tests/integration/components/bs-form-test.js
Comment (dagroe) [reply tests/integration/components/bs-form-test.js:324]: added test for success validation

File: tests/integration/components/bs-form-test.js
Comment (dagroe) [reply tests/integration/components/bs-form-test.js]: same as above: added test

**Comment by dagroe** [comment]
@jelhan Could you review the requested changes? We could really use this feature. :)

**Comment by simonihmig** [comment]
This looks good to me, and I already approved. Thanks for updating the PR!

There is an issue here though, as CI didn't run on the latest push, idk why? But even if it did, it seems master is currently failing, so I need to make master green again before merging any new PRs...

**Comment by dagroe** [comment]
> There is an issue here though, as CI didn't run on the latest push, idk why? But even if it did, it seems master is currently failing, so I need to make master green again before merging any new PRs...

I updated my branch to the latest master with passing CI. It seems CI doesn't run here since it needs to be approved.

**Comment by jelhan** [comment]
I'm sorry for delayed review. Will trying to do it in this week and getting it shipped with v6.

CI is failing. Do you mind having a look?

**Comment by dagroe** [comment]
> I'm sorry for delayed review. Will trying to do it in this week and getting it shipped with v6.
> 
> CI is failing. Do you mind having a look?

Thanks!

I'll fix CI. Seems there is an extra line break the linter is complaining about...

**Comment by dagroe** [comment]
I updated my branch. CI requires approval again. Fingers crossed it passes this time.

**Review by jelhan — APPROVED** [review]
Looks good to me. Thanks a lot and sorry for the slow feedback! Haven't had much time for OSS in last months.

**Comment by jelhan** [comment]
@dagroe This has been released as part of [`v6.0.0-0`](https://github.com/ember-bootstrap/ember-bootstrap/releases/tag/v6.0.0-0). I'm sorry that it took that long.

**Pull request merged**
